### PR TITLE
fix: turn hasura actions coin amount type to string

### LINKED ISSUE #325: [hasura actions] turn all the response coin type to string?
## Feature description 

Given that sifchain is 18 exponents, it's easy to encounter "range exceeds int64" while implementing hasura actions. I then aborted sdk.Coin as response type and created custom type of Coin for actions with turning the Amount field into string which solves the issue.

Should we turn the amount field to string for all the coin responses ?

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #325

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [x] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [x] reviewed tests and test coverage
- [x] manually tested (if applicable)

### CODE REVIEW

**Review by MonikaCat — CHANGES_REQUESTED** [review]
I think we can also update Coins object to `[]Coin` type in `Delegation` struct, can you add it pls?

File: cmd/actions/types/utils.go [hunk 1]
```diff
@@ -1,5 +1,29 @@
 package types
 
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
 type GraphQLError struct {
 	Message string `json:"message"`
 }
+
+func ConvertSdkCoins(s sdk.Coins) []Coin {
```
Comment (MonikaCat) [inline cmd/actions/types/utils.go]: This seems a bit redundant. There are methods `NewDbDecCoin`, `NewDbDecCoins `, `NewDbCoin` 
 and `NewDbCoins` that convert the `Amount` value to string, can we use them instead?

**Review by MonikaCat — APPROVED** [review]

**Review by MonikaCat — APPROVED** [review]
tested, LGTM

**Review by rmontagnin — APPROVED** [review]
utACK

**Pull request merged**